operator/v1/types_ingress.go

```diff
@@ -68,6 +68,7 @@ type IngressControllerSpec struct {
 	//
 	// If empty, defaults to ingress.config.openshift.io/cluster .spec.domain.
 	//
+	// +kubebuilder:validation:XValidation:rule="(has(oldSelf) && self == oldSelf) || self.matches('^([a-z0-9]([-a-z0-9]{0,61}[a-z0-9])?)'+'(\\.[a-z0-9]([-a-z0-9]{0,61}[a-z0-9])?)*$')",message="domain must consist of lowercase alphanumeric characters, '-' or '.', and each label must start and end with an alphanumeric character and not exceed 63 characters"
```

Comment:
Which releases are you planning to fix this bug in? With ratcheting validation in the API server you shouldn't need `(has(oldSelf) && self == oldSelf)` anymore, but we can test that

Reply:
Validation ratcheting became enabled by default from 4.17, with this, you only need to worry about ractheting explicitly if you have some edge case requirements, in which case you'd normally use `optionalOldSelf` as well